### PR TITLE
A simple battery app.

### PR DESCRIPTION
**Pull request by craftit**

### CODE REVIEW

**Review by kliment — CHANGES_REQUESTED** [review]
Charging status logic is the wrong way around. Button detection should trigger when voltage read is below UVLO threshold (will read as close to zero) and not attempt to report battery status in that case.

File: modules/battery/__init__.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        self.adc.width(ADC.WIDTH_12BIT)
+        self.timer = None
+
+    def on_activate(self):
+        super().on_activate()
+        # Make sure the pullup is disabled.
+        # We should probably disable interrupts from the pin too here.
+        self.pin.init(self.pin.IN, None)
+        window = self.window
+        window.cls()
```
Comment (tomsci) [inline modules/battery/__init__.py:86]: minor: `cls()` should be getting called as a result of the `super().on_activate()` call (which calls `TextWindow.redraw()`) so this shouldn't be needed here?

**Comment by craftit** [comment]
The raw voltage should probably be removed in the final version, but it is helpful for testing so I'll leave it in.

File: modules/battery/__init__.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        super().on_start()
+        self.adc = ADC(self.pin)
+        self.adc.atten(ADC.ATTN_11DB)
+        self.adc.width(ADC.WIDTH_12BIT)
+        self.timer = None
+
+    def on_activate(self):
+        super().on_activate()
+        # Make sure the pullup is disabled.
+        # We should probably disable interrupts from the pin too here.
```
Comment (tomsci) [inline modules/battery/__init__.py:90]: Unless you've set an interrupt on BUTTON_A yourself (by calling `Buttons.on_press()` or `on_up_down()`), the app switching mechanism will guarantee nobody else has an active interrupt on it between calls to on_activate() and on_deactivate().

**Review by tomsci — APPROVED** [review]
App seems to work nicely for me

**Review by kliment — APPROVED** [review]

**Pull request merged**